cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -1312,6 +1324,42 @@ private String getHiveDatabase(String namespace) {
     return sessionConf;
   }
 
+  /**
+   * Updates the token store to be used for the hive job, based upon the Explore container's credentials.
+   * This is because twill doesn't update the container_tokens on upon token refresh.
+   * See: https://issues.apache.org/jira/browse/TWILL-170
+   */
+  private void updateTokenStore() throws IOException, ExploreException {
+    String hadoopTokenFileLocation = System.getenv(UserGroupInformation.HADOOP_TOKEN_FILE_LOCATION);
+    if (hadoopTokenFileLocation == null) {
+      LOG.warn("Skipping update of token store due to failure to find environment variable '{}'.",
+               UserGroupInformation.HADOOP_TOKEN_FILE_LOCATION);
+      return;
+    }
+
+    File credentialsFile = new File(hadoopTokenFileLocation);
+
+    FileAttribute<Set<PosixFilePermission>> originalPermissionAttributes =
+      PosixFilePermissions.asFileAttribute(java.nio.file.Files.getPosixFilePermissions(credentialsFile.toPath()));
+
+    File tmpFile = java.nio.file.Files.createTempFile(credentialsFile.getParentFile().toPath(),
+                                                      "credentials.store", null,
+                                                       originalPermissionAttributes)
+      .toFile();
```

Comment:
You don't have to `toFile`. The `createTempFile` returns you a `Path` object and you can use `Files.newOutputStream` for writing and `Files.move` to do the moving.